File: api/src/main/java/com/api/service/InvoiceProductService.java
package com.api.service;

import com.api.dto.InvoiceProductDTO;
import com.api.dto.InvoiceProductIndexDTO;
import com.api.dto.ProductStockDTO;
import com.api.model.InvoiceProduct;

import java.text.ParseException;
import java.util.List;

public interface InvoiceProductService {

    InvoiceProduct dtoToEntity(InvoiceProductDTO invoiceProductDTO);

    List<InvoiceProductDTO> findAll();

    InvoiceProduct get(Long id);

    InvoiceProduct save(InvoiceProductDTO invoiceProductDTO);

    //InvoiceProduct update(InvoiceProductDTO invoiceProductDTO);

    void delete(Long id);

    List<InvoiceProductIndexDTO> index(InvoiceProductIndexDTO invoiceProductIndexDTO);

    List<InvoiceProduct> findByProductIdAndActionDate(ProductStockDTO productStockDTO) throws ParseException;
}

File: front-end/src/app/np/invoiceProduct/invoiceProduct.service.js
(function() {

    angular.module('app.np.invoiceProduct').service('invoiceProductService', invoiceProductService);

    function invoiceProductService(invoiceProductFactory) {

        return {
            getInvoiceProducts: getInvoiceProducts,
            getInvoiceProduct: getInvoiceProduct,
            saveInvoiceProduct: saveInvoiceProduct,
            deleteInvoiceProduct: deleteInvoiceProduct,
            indexInvoiceProduct: indexInvoiceProduct
        }

        function getInvoiceProducts() {
            return invoiceProductFactory.list({});
        }

        function getInvoiceProduct(id) {
            return invoiceProductFactory.get({id}, null);
        }

        function saveInvoiceProduct(invoiceProduct) {
            return invoiceProductFactory.save({},invoiceProduct);
        }

        function deleteInvoiceProduct(id) {
            return invoiceProductFactory.delete({id}, null);
        }

        function indexInvoiceProduct(search) {
            return invoiceProductFactory.index({}, search);
        }

    }

})();